docs/getting-started-guides/locally.md

```diff
@@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
 
 Congratulations!
 
+### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
+
+To test highly available apiservers, you can modify local-up-cluster.sh as follows.
+
+1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
+
+2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
```

Comment:
The cert is *not* going to list 127.0.0.1 as a SAN, so kubelet can't
validate it if connecting via the nodeport.

On Wed, Sep 23, 2015 at 4:30 PM, jay vyas <notifications@github.com> wrote:

> In docs/getting-started-guides/locally.md
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40271139>
> :
>
> > @@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
> >
> >  Congratulations!
> >
> > +### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
> > +
> > +To test highly available apiservers, you can modify local-up-cluster.sh as follows.
> > +
> > +1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
> > +
> > +2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
>
> To be clear on certs... The kubeconfig passed to kubelet handles
> validation. Not sure what the debate is there? cert generation and tooling
> will want to create certs for the kubelets to allow them to authenticate
> their connection to apiserver rather than using insecure port.
> right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40271139>.
>